src/Templates.php

```diff
@@ -0,0 +1,358 @@
+        }
+
+        return $instance;
+    }
+
+    public function getLocalIncludePath()
+    {
+        return $this->localIncludePath;
+    }
+
+    public function setLocalIncludePath($localIncludePath)
+    {
+        $this->localIncludePath = $localIncludePath;
+        return $this;
+    }
+
+    public function getTemplateFile()
+    {
+        return $this->getCache();
+    }
+
+    /**
+     * Attempts to connect to the template server and grabs the latest cache of the
+     * template (.tpl) file. Set options for Cache_Lite in self::$options['cache']
+     *
+     * @return string
+     */
+    protected function getCache()
+    {
+        $localIncludePath = $this->getLocalIncludePath();
+
+        if (!$localIncludePath && !empty(static::$options['templatedependentspath'])) {
+            $localIncludePath = static::$options['templatedependentspath'];
+        }
+
+        $cache = static::getCachingService();
+        $cacheKey = static::VERSION . $localIncludePath . $this->template;
+
+        // Test if there is a valid cache for this template
+        if ($data = $cache->get($cacheKey)) {
+            return $data;
+        }
+
+        $templateFile = $this->getTemplatePath();
+
+        if (!file_exists($templateFile)) {
+            return '';
+        }
+
+        if ($data = file_get_contents($templateFile)) {
+            $data = static::makeIncludeReplacements($data, $localIncludePath);
+            $cache->save($data, $cacheKey);
+        }
+
+        return $data;
+    }
+
+    protected function getTemplatePath()
+    {
+        return static::getDataDir() . DIRECTORY_SEPARATOR . self::TEMPLATE_CACHE_DIR . DIRECTORY_SEPARATOR .
+            static::LOCAL_NAME . DIRECTORY_SEPARATOR . $this->template;
+    }
+
+    protected function appendToHead($value)
+    {
+        $head = $this->getRegion('head');
+
+        if (!$head) {
+            return $this;
+        }
+
+        $head->setValue($head->getValue() . $value);
+        return $this;
+    }
+
+    /**
+     * Add a link within the head of the page.
+     *
+     * @param string $href       URI to the resource
+     * @param string $relation   Relation of this link element (alternate)
+     * @param string $relType    The type of relation (rel)
+     * @param array  $attributes Any additional attribute=>value combinations
+     * @return self
+     */
+    public function addHeadLink($href, $relation, $relType = 'rel', array $attributes = array())
+    {
+        $attributes[$relType] = $relation;
+        $attributes['href'] = $href;
+        $element = static::generateElement('link', $attributes, false) . PHP_EOL;
+        return $this->appendToHead($element);
+    }
+
+    /**
+     * Add a (java)script to the page.
+     *
+     * @param string $url  URL to the script
+     * @param string $type Type of script text/javascript
+     * @return self
+     */
+    public function addScript($url, $type = '')
```

Comment:
I'm not looking to _add_ to the API at this point. Just maintaining b/c. Adding to the end of the body is not currently supported by the framework.